### ISSUE TITLE
How to compute the loss function curve from the train.py in image classification

### ISSUE THREAD

**Issue by danisomo** [dusty-nv/jetson-inference#851]
Hello, first of all, i am doing my final project of Master's Degree and i had some issues which i can't solve.
My project is about the jetbot recognising four classes images in a long corridor which are right, left, forward and obstacle and making it move depending on the returned class given by a DNN.
Having said this, i tried using different networks provided such as googlenet, alexnet, resnet18, resnet50, etc. but the results with the pretained models are not very good.
I guess this happens because the classes with which they have trained those pretrained models of these networks have nothing to do with the classes that i am trying to classify, so it does not work, specially with classes right and left, so the next step i followed was to deactivate the flag pretrained in the script of training called train.py writting False in the default value in order to avoid the transfer learning and training the network right from the start. 
Then, i have been analyzing the code and watching the results that appear in terminal but the loss value does not seem very real because it is not appreciated how the losses are decreasing. 
Can anybody help me or explain me why i cannot see the losses converge to a low value as it should happen? The values of the epoch one are very similar in order of magnitude to the last epoch as can be seen in the next pictures:

![image](https://user-images.githubusercontent.com/58220236/102565857-94e6c080-40de-11eb-81c5-6af816d5ab56.png)
![epoch45](https://user-images.githubusercontent.com/58220236/102565881-9fa15580-40de-11eb-8191-2880ef779e00.png)

My teacher told me that it is absolutely necessary to represent the loss function curve in the report in order to justify how the network is learning and not only the plot of the accuracy (this one can be plot without difficulties)

**Comment by dusty-nv** [comment]
Those loss values are printed for each 10 batches, and vary from batch to batch.   The per-batch accuracy varies up & down also. 
 What you may want to do is print out the loss after the validation step at the end of each epoch.   You could print that here:

https://github.com/dusty-nv/pytorch-classification/blob/15c7d51cd6a1271e5ee6959c3f91fb95b6b54ff0/train.py#L395

The validation loss is tracked in the `losses` object, and you could print them just like Acc@1 and Acc@5

**Comment by danisomo** [comment]
Thank you very much @dusty-nv 
For tracking the losses in validation and test is like you said:

![train_print_data](https://user-images.githubusercontent.com/58220236/102818075-37e65580-43d1-11eb-95b1-124970b35cba.png)
![val_print](https://user-images.githubusercontent.com/58220236/102818077-39178280-43d1-11eb-8722-438f6d9f1fd9.png)

In order to plot those values i put them in a list as seen in previous photos. Then i call the function plot at the end of the program as seen in next picture:

![plor function](https://user-images.githubusercontent.com/58220236/102818279-97446580-43d1-11eb-8ef5-51b668150bea.png)

The plot function is defined like this:

![plor function](https://user-images.githubusercontent.com/58220236/102818368-c22eb980-43d1-11eb-81cc-ac78c3ec92c4.png)

At the end i get the results of acc1, acc5 and losses each time an epoch occurs in train and validation.

And when the last epoch is executed i got this two plots which i think they look very good. 

![Acc@1 (1)](https://user-images.githubusercontent.com/58220236/102818487-06ba5500-43d2-11eb-90b1-ee54edcdbfee.png)
![loss (1)](https://user-images.githubusercontent.com/58220236/102818488-06ba5500-43d2-11eb-8a48-6c3c8ceade58.png)

Sorry for copying things that should be known but just wanted to know if the values are well obtained for the plots and if the plots look pretty good? 

Thanks you beforehand
